### PR TITLE
Wave function compact visualization

### PR DESCRIPTION
**Pull request by davibinco**

### CODE REVIEW

File: src/tequila/wavefunction/qubit_wavefunction.py [hunk 1]
```diff
@@ -365,15 +365,18 @@ def normalize(self, inplace: bool = False) -> QubitWaveFunction:
     # because the __mul__ implementation of the number tries to perform some sort of array
     # operation.
     def length(self):
-        return sum(1 for (k, v) in self.raw_items() if abs(v) > 1e-6)
+        return sum(1 for _ in self.raw_items())
```
Comment (ohuettenhofer) [inline src/tequila/wavefunction/qubit_wavefunction.py:368]: Is this change on purpose? I'm guessing you branched off before https://github.com/tequilahub/tequila/pull/375?

File: src/tequila/wavefunction/qubit_wavefunction.py [hunk 2]
```diff
@@ -365,15 +365,18 @@ def normalize(self, inplace: bool = False) -> QubitWaveFunction:
...
 
     def __repr__(self):
         result = str()
         for index, coeff in self.items():
             index = index.integer
             if self.numbering == BitNumbering.LSB:
                 index = reverse_int_bits(index, self._n_qubits)
-            result += f"{coeff} |{index:0{self._n_qubits}b}> "
+            if np.isclose(coeff.imag,0.0):
```
Comment (ohuettenhofer) [inline src/tequila/wavefunction/qubit_wavefunction.py:376]:
```suggestion
            if np.isclose(coeff.imag, 0.0):
```
Missing space

File: src/tequila/wavefunction/qubit_wavefunction.py [hunk 3]
```diff
@@ -365,15 +365,18 @@ def normalize(self, inplace: bool = False) -> QubitWaveFunction:
...
+                result += f"{coeff.real:+2.4f} |{index:0{self._n_qubits}b}> "
+            else:
+                result += f"({coeff.real:+2.4f} + {coeff.imag:+2.4f}i) |{index:0{self._n_qubits}b}>"
```
Comment (ohuettenhofer) [inline src/tequila/wavefunction/qubit_wavefunction.py:379]:
```suggestion
                result += f"({coeff.real:+2.4f} + {coeff.imag:+2.4f}i) |{index:0{self._n_qubits}b}> "
```
Missing space

File: src/tequila/wavefunction/qubit_wavefunction.py (hunk 1 above)
Comment (davibinco) [reply src/tequila/wavefunction/qubit_wavefunction.py:368]: I did it but I overwrote it by mistake. Should I make a new one

Comment (ohuettenhofer) [reply src/tequila/wavefunction/qubit_wavefunction.py:368]: A new commit? Yes, or amend this one to remove the change. (Unless I misunderstood and you mean the change was on purpose?)

Comment (davibinco) [reply src/tequila/wavefunction/qubit_wavefunction.py]: No, it wasn't on purpose. I made a new commit.

**Comment by kottmanj** [comment]
Good to go?
@davibinco, @ohuettenhofer

**Comment by ohuettenhofer** [comment]
I think so, yes.

**Comment by kottmanj** [comment]
merging. No idea what the issue is with the psi4 installation in the tests (works locally)
